src/super_gradients/training/utils/media/image.py

```diff
@@ -44,31 +45,49 @@ def generate_image_loader(images: Union[List[ImageSource], ImageSource]) -> Iter
         - List:             A list of images of any of the above types.
 
     :param images:  Single image or a list of images of supported types.
-    :return:        Generator of images as numpy arrays. If loaded from string, the image will be returned as RGB.
+    :return:        Generator of images as numpy arrays (H, W, C). If loaded from string, the image will be returned as RGB.
     """
     if isinstance(images, str) and os.path.isdir(images):
         images_paths = list_images_in_folder(images)
         for image_path in images_paths:
             yield load_image(image=image_path)
-    elif isinstance(images, (list, Iterator)):
+    elif _is_4d_array(images):
+        warnings.warn(
```

Comment:
suggestion: just warn once (could be the case that people would iterate through some data loader and call predict on 4d tensors multiple times leaving a mess here).

Reply:
Actually, warnings.warn works exactly that.
See the docs at http://docs.python.org/2/library/warnings.html:

Repetitions of a particular warning for the same source location are typically suppressed.